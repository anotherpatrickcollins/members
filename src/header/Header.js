import React from 'react';
import { Link } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import logo from '../images/cfd-circle-icon-white.png';
import userIsAdmin from '../utils/userIsAdmin';
import AuthenticationContext from '../utils/authentication/authContext';
import AuthService from '../utils/authentication/authService';
import './Header.css';

const Header = () => {
  return (
    <AuthenticationContext.Consumer>
      {context => (
        <AppBar position="fixed">
          <Toolbar>
            <Link to="/">
              <img
                className="Header-logo"
                src={logo}
                alt="code for denver logo"
              />
            </Link>
            <Grid container justify="space-between" alignItems="center">
              {context.isAuthenticated() ? (
                <Grid item>
                  <Link className="Header-link" to="/volunteers">
                    All Users
                  </Link>
                  <Link className="Header-link" to="/projects">
                    All Projects
                  </Link>
                  {userIsAdmin(context.authData.role) && (
                    <Link className="Header-link" to="/admin">
                      Admin Resources
                    </Link>
                  )}
                </Grid>
              ) : (
                <Grid item>
                  <Button color="secondary" onClick={() => AuthService.login()}>
                    Login
                  </Button>
                  <Button
                    color="secondary"
                    onClick={() => AuthService.signUp()}
                  >
                    Signup
                  </Button>
                </Grid>
              )}
            </Grid>
          </Toolbar>
        </AppBar>
      )}
    </AuthenticationContext.Consumer>
  );
};

export default Header;
