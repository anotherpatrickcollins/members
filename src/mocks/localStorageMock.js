import {
  ACCESS_TOKEN_KEY,
  BEARER_TOKEN,
  USER_ID,
  EXPIRES_AT_KEY
} from '../constants/storageKeys';

export default {
  getItem: jest.fn(key => {
    switch (key) {
      case ACCESS_TOKEN_KEY:
        return 'accessToken';
      case EXPIRES_AT_KEY:
        return 2535419869506;
      case USER_ID:
        return 'mocked-test-user-id';
      case BEARER_TOKEN:
        return 'bearerToken';
      default:
        return null;
    }
  }),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn()
};
