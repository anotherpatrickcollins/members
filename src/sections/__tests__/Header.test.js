import React from 'react';
import { queryByText, getByText, waitForElement } from 'react-testing-library';
import Header from '../Header';
import { mountWithAuth, mountWithContext } from '../../testUtils';
import localStorageMock from '../../mocks/localStorageMock';

const mockUser = {
  role: 'USER'
};

const mockAdminUser = {
  role: 'ADMIN'
};

describe('Header', () => {
  describe('if the user is logged in', () => {
    it('should not show the login button', () => {
      const { queryByText } = mountWithAuth(
        <Header isAuthenticated={true} user={mockUser} />
      );
      expect(queryByText('Log In')).toBeNull();
    });

    describe('and the user is not admin', () => {
      it('should show the correct navigation links', async () => {
        const { getByText, queryByText, container } = mountWithAuth(
          <Header isAuthenticated={true} user={mockUser} />
        );

        await waitForElement(() => getByText('All Users'));

        expect(getByText('All Users')).toBeDefined();
        expect(getByText('All Projects')).toBeDefined();
        expect(queryByText('Admin Resources')).toBeNull();
        expect(container).toMatchSnapshot();
      });
    });

    describe('and the user is admin', () => {
      it('should show the correct navigation links', () => {
        const { getByText, container } = mountWithAuth(
          <Header isAuthenticated={true} user={mockAdminUser} />,
          {
            routes: ['/']
          }
        );

        expect(getByText('All Users')).toBeDefined();
        expect(getByText('All Projects')).toBeDefined();
        expect(getByText('Admin Resources')).toBeDefined();
        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('if the user is not logged in', () => {
    beforeAll(() => {
      global.localStorage.getItem = jest.fn(() => null);
    });

    afterAll(() => {
      global.localStorage.getItem = localStorageMock.getItem;
    });

    it('not should show any navigation links', () => {
      const { queryByText, container } = mountWithContext(<Header />, {
        routes: ['/']
      });

      expect(queryByText('All Users')).not.toBeDefined();
      expect(queryByText('All Projects')).not.toBeDefined();
      expect(queryByText('Admin Resources')).not.toBeDefined();
      expect(container).toMatchSnapshot();
    });

    it('should show the login button', () => {
      const { getByText, container } = mountWithContext(
        <Header isAuthenticated={false} />,
        {
          routes: ['/']
        }
      );

      expect(getByText('Log In')).toBeDefined();
      expect(container).toMatchSnapshot();
    });
  });
});
