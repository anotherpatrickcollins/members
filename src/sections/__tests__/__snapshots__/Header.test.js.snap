// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header if the user is logged in and the user is not admin should show the correct navigation links 1`] = `
<div>
  <header
    class="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-7 mui-fixed"
  >
    <div
      class="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <a
        href="/"
      >
        <img
          class="Header-logo"
          src="cfd-circle-icon-white.png"
          alt="code for denver logo"
        />
      </a>
      <div
        class="MuiGrid-container-38 MuiGrid-align-items-xs-center-46 MuiGrid-justify-xs-space-between-57"
      >
        <div
          class="MuiGrid-item-39"
        >
          <a
            class="Header-link"
            href="/volunteers"
          >
            All Users
          </a>
          <a
            class="Header-link"
            href="/projects"
          >
            All Projects
          </a>
        </div>
        <div
          class="MuiGrid-item-39"
        />
      </div>
    </div>
  </header>
</div>
`;

exports[`Header if the user is not logged in should show the login button 1`] = `
<div>
  <header
    class="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-7 mui-fixed"
  >
    <div
      class="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <a
        href="/"
      >
        <img
          class="Header-logo"
          src="cfd-circle-icon-white.png"
          alt="code for denver logo"
        />
      </a>
      <div
        class="MuiGrid-container-38 MuiGrid-align-items-xs-center-46 MuiGrid-justify-xs-space-between-57"
      >
        <div
          class="MuiGrid-item-39"
        >
          <button
            tabindex="0"
            class="MuiButtonBase-root-151 MuiButton-root-129 MuiButton-textSecondary-132 MuiButton-flat-133 MuiButton-flatSecondary-135"
            type="button"
          >
            <span
              class="MuiButton-label-130"
            >
              Log In
            </span>
            <span
              class="MuiTouchRipple-root-154"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
